File: CSharpFundamentals/CSharpAdvanced/SetsAndDictionaries_Lab/ParkingLot.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ParkingLot
{
    class ParkingLot
    {
        static void Main(string[] args)
        {
            HashSet<string> set = new HashSet<string>();
            string input = Console.ReadLine();
            while (input != "END")
            {
                var splitted = input
                    .Split(", ", StringSplitOptions.RemoveEmptyEntries);
                string direction = splitted[0];
                string car = splitted[1];
                if (direction == "IN")
                {
                    set.Add(car);
                }
                else
                {
                    set.Remove(car);
                }

                input = Console.ReadLine();
            }
            if (set.Count == 0)
            {
                Console.WriteLine("Parking Lot is Empty");
                return;
            }
            set.ToList().ForEach(s => Console.WriteLine(s));
        }
    }
}

File: JavaFundamentals/JavaAdvanced/Sets-And-Maps-Lab-2/src/ParkingLot.java
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class ParkingLot {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        Set<String> parkingLot = new HashSet<>();

        while (!input.equals("END")) {
            String[] splitted = input.split(", ");
            String action = splitted[0];
            String carNumber = splitted[1];
            if (action.equals("IN")) {
                parkingLot.add(carNumber);
            } else {
                parkingLot.remove(carNumber);
            }
            input = scanner.nextLine();
        }

        if (parkingLot.isEmpty()) {
            System.out.println("Parking Lot is Empty");
        } else {
            parkingLot.forEach(System.out::println);
        }
    }
}
